pkg/ruler/ruler_test.go

```diff
@@ -623,39 +687,56 @@ func TestGetRules(t *testing.T) {
 
 			// Sync rules on each ruler.
 			for _, r := range rulerAddrMap {
-				r.syncRules(ctx, nil, rulerSyncReasonInitial, true)
+				// Use rulerSyncReasonPeriodic to ensure rules are distributed among ACTIVE rulers
+				r.syncRules(ctx, nil, rulerSyncReasonPeriodic, true)
```

Comment:
TODO: use rulerSyncReasonPeriodic for ACTIVE rulers, otherwise reason initial to properly simular what really happens